src/structures/guild.ts

```diff
@@ -328,18 +331,137 @@ export class Guild extends Base {
       if (!this.unavailable) resolve(this)
       const listener = (guild: Guild): void => {
         if (guild.id === this.id) {
-          this.client.removeListener('guildLoaded', listener)
+          this.client.off('guildLoaded', listener)
           resolve(this)
         }
       }
       this.client.on('guildLoaded', listener)
       setTimeout(() => {
-        this.client.removeListener('guildLoaded', listener)
+        this.client.off('guildLoaded', listener)
         reject(Error("Timeout. Guild didn't arrive in time."))
       }, timeout)
     })
   }
 
+  /** Attach an integration object from the current user to the guild. */
+  async createIntegration(id: string, type: string): Promise<Guild> {
+    await this.client.rest.api.guilds[this.id].integrations.post({ id, type })
+    return this
+  }
+
+  /** Modify the behavior and settings of an integration object for the guild. */
+  async editIntegration(
+    id: string,
+    data: {
+      expireBehavior?: number | null
+      expireGracePeriod?: number | null
+      enableEmoticons?: boolean | null
+    }
+  ): Promise<Guild> {
+    await this.client.rest.api.guilds[this.id].integrations[id].patch({
+      expire_behaviour: data.expireBehavior,
+      expire_grace_period: data.expireGracePeriod,
+      enable_emoticons: data.enableEmoticons
+    })
+    return this
+  }
+
+  /** Delete the attached integration object for the guild. Deletes any associated webhooks and kicks the associated bot if there is one. */
+  async deleteIntegration(id: string): Promise<Guild> {
+    await this.client.rest.api.guilds[this.id].integrations[id].delete()
+    return this
+  }
+
+  /** Sync an integration. */
+  async syncIntegration(id: string): Promise<Guild> {
+    await this.client.rest.api.guilds[this.id].integrations[id].sync.post()
+    return this
+  }
+
+  /** Returns the widget for the guild. */
+  async getWidget(): Promise<GuildWidgetPayload> {
+    return this.client.rest.api.guilds[this.id]['widget.json'].get()
```

Comment:
You can ignore this but converting things into camelCase will be good

Reply:
A separate structure for this would be overkill 🤔

Reply:
Yeah but it matches with our linter style and doesn't make code slower